codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -604,8 +604,8 @@ public void onUnknownFrame(
                 // Ignore unknown frames on connection stream, for example: HTTP/2 GREASE testing
                 return;
             }
-            onHttp2Frame(ctx, newHttp2UnknownFrame(frameType, streamId, flags, payload.retain())
-                .stream(requireStream(streamId)));
+            Http2FrameStream stream = requireStream(streamId);
+            onHttp2Frame(ctx, newHttp2UnknownFrame(frameType, streamId, flags, payload.retain()).stream(stream));
```

Comment:
fwiw, this was fine before #14720. I kind of liked the pattern as it made the `.retain()` call the last thing that happened.

Reply:
See below.